app/controllers/sessions_controller.rb

```diff
@@ -4,11 +4,20 @@ class SessionsController < ApplicationController
   def create
     session[:user_token] = auth_hash.fetch(:credentials).fetch(:token)
 
-    redirect_to root_url
+    redirect_to redirect_url
   end
 
   private
 
+  def redirect_url
+    custody_suite_root_url if user_role == "cso"
+  end
+
+  # TODO: This is hacky
+  def user_role
+    @user_role ||= current_user.organisations.map { |o| o["roles"] }.flatten.uniq.first
```

Comment:
Ok, agree. But can you please use `current_user.roles.uniq.first` instead then? Because if you map the organisations like that, you may pick a role from a different organisation, than the `current valid` one.